warehouse/accounts/forms.py

```diff
@@ -95,7 +95,7 @@ def __init__(self, *args, check_password_metrics_tags=None, **kwargs):
         super().__init__(*args, **kwargs)
 
     def validate_password(self, field):
-        userid = self.user_service.find_userid(self.username.data)
+        userid = self.user_service.find_userid(self._get_username())
```

Comment:
You could probably use the `field` parameter directly here to avoid this `_get_username` helper.

Reply:
I believe in this case `field` is the `password` field not the username. This mixin expects the instance to have a `username` attribute of type field, which it does most of the time as it's usually paired with the `UsernameMixin`, but in this case it's not, so I set it as a [string instance variable](https://github.com/pypa/warehouse/pull/6408/files#diff-19b7c365d44b94a6782decdcdfeee79aR96).

Reply:
Ah, sorry, misinterpreted the context.